tests/attestation_plugin/plugin/tests.c

```diff
@@ -308,6 +304,62 @@ static void _test_verify_evidence_fail()
     OE_TEST(oe_free_endorsements(endorsements) == OE_OK);
 }
 
+static void _test_custom_claims_serde()
+{
+    printf("====== begin  _test_custom_claims_serde\n");
+    uint8_t* claims_out = NULL;
+    size_t claims_size_out = 0;
+    oe_claim_t* claims1 = NULL;
+    size_t claims_len = 0;
+
+    oe_claim_t custom_claims[2] = {{.name = CLAIM1_NAME,
+                                    .value = (uint8_t*)CLAIM1_VALUE,
+                                    .value_size = sizeof(CLAIM1_VALUE)},
+                                   {.name = CLAIM2_NAME,
+                                    .value = (uint8_t*)CLAIM2_VALUE,
+                                    .value_size = sizeof(CLAIM2_VALUE)}};
+
+    printf("====== _test_custom_claims_serde call oe_claims_serialize\n");
+    OE_TEST(
+        oe_serialize_custom_claims(
+            custom_claims, 2, &claims_out, &claims_size_out) == OE_OK);
+
+    printf("====== _test_custom_clims_serde call oe_claims_deserialize\n");
+    OE_TEST(
+        oe_deserialize_custom_claims(
```

Comment:
I'd like to see a more realistic test: one function calls oe_get_evidence with a list of custom claims, and another function calls oe_verify_evidence and verifies the deserialized custom claims.

Reply:
I think the test case should focus on testing serialization/de-serialization. What happens inside oe_get_evidence() and oe_verify_evidence() is not to be covered in this test case.

Reply:
I disagree. The sole purpose of these helper functions is to pass/parse the buffer to/from oe_get_evidence/oe_verify_evidence. We should verify the helper functions end-to-end.

Reply:
The role of custom claims in the process of evidence validation:
[custom claims] -> (1. serialize) -> [flat custom claims buffer] -> (2. oe_get_evidence) -> [evidence] -> (3. oe_verify_evidence) -> [flag custom claim buffer] -> (4. de-serialize) -> [custom claims]

The test cases in this PR should focus on input/output of (1. serialize) and (4. de-serialize).

Issue #3389 will cover the test cases for input/output of (2. oe_get_evidence) and (3. oe_verify_evidence).